File: App/pidrei-ng-app/src/shared/classes/device-temperature.ts
export class DeviceTemperature {
    public temperature: number;
    public offset: number;

    constructor(initialValue: { temperature?: number, offset?: number } = { temperature: 0, offset: 0 }) {
        this.temperature = initialValue.temperature;
        this.offset = initialValue.offset;
    }

    public getTemperature(): number {
        return this.temperature + this.offset;
    }
}

File: FritzBox/Model/DeviceTemperature.cs
using System.Xml.Serialization;
using Newtonsoft.Json;

namespace RaspPiTest.FritzBox.Model
{
    [JsonObject(MemberSerialization.OptIn)]
    public class DeviceTemperature
    {
        [XmlElement("celsius")]
        public int CelsiusInternal { get; set; }
        [JsonProperty("temperature")] public float Temperature => CelsiusInternal / 10f;

        [XmlElement("offset")]
        public int OffsetInternal { get; set; }
        [JsonProperty("offset")] public float Offset => OffsetInternal / 10f;
    }
}